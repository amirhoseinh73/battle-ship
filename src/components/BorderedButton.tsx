import React from "react"
import { Text, StyleSheet, Pressable, GestureResponderEvent, View } from "react-native"

type BtnProps = {
  title: string
  onPress?: (event: GestureResponderEvent) => void
} & Record<string, unknown>

const BorderedButton = function (props: BtnProps) {
  const { onPress, title = "Start", ...rest } = props

  // console.log(rest)

  return (
    <View style={styles.container}>
      <Pressable style={styles.button} onPress={onPress}>
        <Text style={styles.text}>{title}</Text>
      </Pressable>
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: "#ddd",
    padding: 10,
    marginBottom: "auto",
    borderRadius: 10,
  },
  button: {
    alignItems: "center",
    justifyContent: "center",
    paddingVertical: 6,
    paddingHorizontal: 22,
    elevation: 0,
    backgroundColor: "inherit",
    borderColor: "#245bdc",
    borderStyle: "dashed",
    borderWidth: 4,
  },
  text: {
    fontSize: 16,
    lineHeight: 21,
    fontWeight: "bold",
    letterSpacing: 0.25,
    color: "white",
  },
})

export default BorderedButton
