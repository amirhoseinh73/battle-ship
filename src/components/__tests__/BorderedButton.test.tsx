import React from "react"
import renderer from "react-test-renderer"
import BorderedButton from "../BorderedButton"

test("renders correctly BorderedButton", () => {
  const btn = renderer.create(<BorderedButton title="test" />).toJSON()
  expect(btn).toMatchSnapshot()
})
