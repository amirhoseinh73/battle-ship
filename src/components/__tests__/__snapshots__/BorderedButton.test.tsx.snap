// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly BorderedButton 1`] = `
<View
  style={
    {
      "backgroundColor": "#ddd",
      "borderRadius": 10,
      "marginBottom": "auto",
      "padding": 10,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "inherit",
        "borderColor": "#245bdc",
        "borderStyle": "dashed",
        "borderWidth": 4,
        "elevation": 0,
        "justifyContent": "center",
        "paddingHorizontal": 22,
        "paddingVertical": 6,
      }
    }
  >
    <Text
      style={
        {
          "color": "white",
          "fontSize": 16,
          "fontWeight": "bold",
          "letterSpacing": 0.25,
          "lineHeight": 21,
        }
      }
    >
      test
    </Text>
  </View>
</View>
`;
